qualtran/bloqs/chemistry/trotter.ipynb

```diff
@@ -0,0 +1,752 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nbits = 6\n",
+    "nbits_rij_sq = 2 * nbits + 2\n",
+    "print(f\"L = {2**nbits_rij_sq}\")\n",
+    "hit = True \n",
+    "g = 0\n",
+    "# bit count : range\n",
+    "ranges_a = {}\n",
+    "ranges_b = {}\n",
+    "for l in range(0, 2**nbits_rij_sq):\n",
+    "    k = l.bit_length()\n",
+    "    if k > 2:\n",
+    "        if l & (1 << (k-2)) and not hit:\n",
+    "            hit = True\n",
+    "            print(f\"l = {l:0{nbits_rij_sq}b}, range = [{l:5d}, {l+2**(k-2)-1:5d}], length = {2**(k-2):>4d} {k}\")\n",
+    "            g += 1\n",
+    "        if not l & (1 << (k - 2)) and hit:\n",
+    "            hit = False\n",
+    "            print(f\"l = {l:0{nbits_rij_sq}b}, range = [{l:5d}, {l+2**(k-2)-1:5d}], length = {2**(k-2):>4d} {k}\")\n",
+    "            g += 1\n",
+    "# Add four to account for 0, 1, 2, 3\n",
+    "print(f\"number of distinct regions g = {g + 4}. Toffoli cost g - 2 = {g - 2 + 4}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see by the bit pattern that we can use the $k-1$ st bit to toggle between our two ranges [1, 3/2] and [3/2, 2]. The total Toffoli cost is then just g - 2, which is 26 if $r_{ij}^2$ is stored with 14 bits. Note we can store the coefficients to high precision as there is only a small clifford overhead associated with this. In the fusion paper 15 bits of precision is suggested as sufficient. \n",
+    "\n",
+    "To summarize we:\n",
+    "\n",
+    "1. Fit our function $1/x$ in two intervals [1, 3/2] and [3/2, 2].\n",
+    "2. Given a value of $r_{ij}$, use variably spaced qrom to output the appropriately scaled polynomial coefficients.\n",
+    "3. Evaluate the polynomial.\n",
+    "4. Perform a single step of Newton Raphson.\n",
+    "5. Apply a phase to our wavefunction."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ```Trotter Unitaries```\n",
```

Comment:
please make sure there is only one `<h1>` (i.e. one hash) heading per notebook which serves as the title. Otherwise the doc system won't work (as well)

Reply:
Done.